lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/Pure.hs

```diff
@@ -230,16 +237,33 @@ getWalletTip :: WalletState -> Read.ChainPoint
 getWalletTip = walletTip
 
 rollForwardMany
-    :: NonEmpty (Read.EraValue Read.Block) -> WalletState -> WalletState
-rollForwardMany blocks w = foldl' (flip rollForwardOne) w blocks
+    :: ResolveSlot
+    -> NonEmpty (Read.EraValue Read.Block)
+    -> WalletState
+    -> WalletState
+rollForwardMany resolveSlot blocks w =
+    foldl' (flip $ rollForwardOne resolveSlot) w blocks
 
 rollForwardOne
-    :: Read.EraValue Read.Block -> WalletState -> WalletState
-rollForwardOne (Read.EraValue block) w =
+    :: ResolveSlot
+    -> Read.EraValue Read.Block
+    -> WalletState
+    -> WalletState
+rollForwardOne resolveSlot eblock@(Read.EraValue block) w =
     w
         { walletTip = Read.getChainPoint block
         , utxoHistory = rollForwardUTxO isOurs block (utxoHistory w)
         , submissions = Delta.apply (Sbm.rollForward block) (submissions w)
+        , txHistory =
+            TxHistory.rollForward
+                ( valueTransferFromResolvedTx
+                    . resolveInputs
+                        (getUTxO $ utxoHistory w)
+                )
```

Comment:
I think that this is incorrect when the block contains multiple transactions that depend on each other. 😅 The issue is that each transaction may change the UTxO set, and `getUTxO $ utxoHistory w` is the correct UTxO set only at the beginning of the block.

Could you add the computation of the `ValueTransfer` resolution to `Cardano.Wallet.Deposit.Pure.Balance`?

For this, I recommend to augment the `applyBlock` function to compute the data you need. Also, use `TxId` instead of `Tx era` for resolution; this prevents `TxHistory.rollForward` from attempting to do any computation.

We need to be careful about correctness here, that is why I want to isolate the computation of `ValueTransfer` in `Cardano.Wallet.Deposit.Pure.Balance` and property test the hell out of it there.